beamer/chains.py

```diff
@@ -0,0 +1,106 @@
+from dataclasses import dataclass
+from beamer.typing import ChainId
+
+
+@dataclass(frozen=True)
+class ChainDescriptor:
+    id: ChainId
+    ecosystem_name: str
+    name: str
+    type2: bool = False
+    bedrock: bool = False
+    testnet: bool = False
+    local: bool = False
+
+    @property
+    def supports_type_2_transactions(self) -> bool:
+        return self.type2 and not self.local
+
+    @property
+    def is_testnet(self) -> bool:
+        return self.testnet
+
+    @property
+    def is_bedrock(self) -> bool:
+        return self.bedrock
+
+
+@dataclass(frozen=True)
+class _BeamerConfig:
+    chains: dict[str, dict[str, ChainDescriptor]]
+
+    def get_descriptor(self, chain_id: ChainId) -> ChainDescriptor:
+        for networks in self.chains.values():
+            for descriptor in networks.values():
+                if descriptor.id == chain_id:
+                    return descriptor
+        return ChainDescriptor(id=chain_id, ecosystem_name="default", name="default")
+
+
+BEAMER_CONFIG = _BeamerConfig(
+    chains={
+        "ethereum": {
+            "mainnet": ChainDescriptor(
+                id=ChainId(1), ecosystem_name="ethereum", name="mainnet", type2=True
+            ),
+            "goerli": ChainDescriptor(
+                id=ChainId(5), ecosystem_name="ethereum", name="goerli", type2=True, testnet=True
+            ),
+            "local": ChainDescriptor(
+                id=ChainId(1337), ecosystem_name="ethereum", name="local", type2=True, local=True
+            ),
+        },
+        "optimism": {
+            "mainnet": ChainDescriptor(
+                id=ChainId(10), ecosystem_name="optimism", name="mainnet", type2=True, bedrock=True
+            ),
+            "goerli": ChainDescriptor(
+                id=ChainId(420),
+                ecosystem_name="optimism",
+                name="goerli",
+                type2=True,
+                bedrock=True,
+                testnet=True,
+            ),
+            "local": ChainDescriptor(
+                id=ChainId(901),
+                ecosystem_name="optimism",
+                name="local",
+                type2=True,
+                bedrock=True,
+                local=True,
+            ),
+        },
+        "arbitrum": {
+            "mainnet": ChainDescriptor(
+                id=ChainId(42161), ecosystem_name="arbitrum", name="mainnet"
+            ),
+            "goerli": ChainDescriptor(
+                id=ChainId(421613), ecosystem_name="arbitrum", name="goerli", testnet=True
+            ),
+            "local": ChainDescriptor(
+                id=ChainId(412346), ecosystem_name="arbitrum", name="local", local=True
+            ),
+        },
```

Comment:
Seeing that there's a fair amount of repetition here, how about we use a plain tuple for the entire data, e.g.
```python
_DESCRIPTORS = (
   # Arbitrum
   ChainDescriptor(id=ChainId(42161), ecosystem_name="arbitrum", name="mainnet"),
   ChainDescriptor(id=ChainId(421613), ecosystem_name="arbitrum", name="goerli", testnet=True),
   ChainDescriptor(id=ChainId(412346), ecosystem_name="arbitrum", name="local", local=True),

...
)
_DESCRIPTORS = _prepare_for_lookups_DESCRIPTORS)
```
?

The `_prepare_for_lookups` (sorry, couldn't find a better name) would take the tuple data and convert it in a form better suited to lookups so we can have O(1) behaviour, as mentioned above.